LPS-61319 Users in a User Group do not inherit role permissions

pei-jung opened the pull request:
/cc @jorgeferrer @drewbrokke

liferay-continuous-integration commented:
<h1>Build has started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest(master)/20/">test-portal-acceptance-pullrequest(master)</a></p><h5>For more details click <a href="https://test-1-6.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/20/jenkins-report.html">here</a>.</h5>

topolik commented:
Thanks @pei-jung.

So the issue seems the join with Group entity. The previous join from https://github.com/topolik/liferay-portal/pull/355/files/8d82fa7321a80b69c85c428d5452bd943f5231d7..97fbebbefe1089f7ced79dfbbf4fd721f63d7f37#r69359643 is
```
         INNER JOIN Groups_UserGroups
           ON (Groups_UserGroups.groupId = Group_.groupId)
         INNER JOIN Users_UserGroups
           ON (Users_UserGroups.userGroupId = Groups_UserGroups.userGroupId)
```

and you added a new join (https://github.com/pei-jung/liferay-portal/blob/62ed9a46ff4af841690b2b60f3aa738734ec87e2/portal-impl/src/custom-sql/portal.xml#L325-L329)
```
			INNER JOIN
				Users_UserGroups ON
					(Users_UserGroups.userGroupId = Group_.classPK)
			WHERE
				(Users_UserGroups.userId = ?)
```

Is the old join worth keeping then when it doesn't work? Or did I miss anything?

Thanks!

pei-jung commented:
The previous join

```
 INNER JOIN Groups_UserGroups
     ON (Groups_UserGroups.groupId = Group_.groupId)
 INNER JOIN Users_UserGroups
     ON (Users_UserGroups.userGroupId = Groups_UserGroups.userGroupId)
```

gets the Site group ids that are associated with the user groups.

I added a new join

```
INNER JOIN
    Users_UserGroups ON
        (Users_UserGroups.userGroupId = Group_.classPK)
```

to get the user's UserGroup group id.

`GroupLocalService.getUserGroups` should return all groups associated with the user, which include sites, organizations, and user groups. The issue with the previous join is that it only gets the sites where the usergroup is a member of, but not the UserGroup group. Thus, I added a new join to take into account the usergroup groups. We still want to keep the old join the get all the usergroup site groups.

I just realized we need a test for this. I'm going to close this and resend with a test. Thanks!

pei-jung commented:
PR sent to https://github.com/JorgeFerrer/liferay-portal/pull/756.
